src/main/java/com/maxmind/db/Decoder.java

```diff
@@ -287,7 +287,7 @@ private static boolean decodeBoolean(int size)
         // We look for Object.class because if we're decoding a nested map into
         // a Map, the second level is an Object.class. See Object.class in
         // decodeMapIntoMap().
-        if (cls.equals(Map.class) || cls.equals(Object.class)) {
+        if (cls.isAssignableFrom(Map.class) || cls.equals(Object.class)) {
```

Comment:
If `cls` is a concrete class (e.g., a `ConcurrentHashMap`), won't we have to pass `cls` to `decodeMapIntoMap` and create that class rather than a `HashMap`?

Reply:
Good point. I think I've done something similar now.